src/main/java/hudson/plugins/performance/constraints/blocks/PreviousResultsBlock.java

```diff
@@ -16,8 +16,11 @@
     /**
      * True: relative constraint includes a user defined number of builds into the evaluation False:
      * relative constraint includes all builds that have taken place in an user defined time frame
+     * BASELINE: relative constraint includes baseline build defined in the PerformancePublisher settings.
+     * True and false are retained for backward compatibility.
      */
-    private boolean choicePreviousResults;
+    public static final String PREVIOUS = "true", TIMEFRAME = "false", BASELINE = "BASELINE";
+    private String choicePreviousResults = TIMEFRAME; // keep field name for backward compatibility
```

Comment:
What about backward compatibility? 
Does this class will load from file storage without any problems? or this class should have `readResolve` method for migrate boolean field to new String? https://wiki.jenkins.io/display/JENKINS/Hint+on+retaining+backward+compatibility

Reply:
Thanks for the tip about `readResolve`. I tested it and it does work without.
Because there are no new fields added that would need a non-trivial initial value when hydrating config back from XML, nor renamed fields that would need a value copied from the old field.
Otherwise, when creating a new config/pipeline, the `@DataBoundConstructor` will be used and the String value gets initialized with a correct (non-null) value.

Reply:
Great! Thanks!